app/models/manageiq/providers/cloud_manager.rb

```diff
@@ -31,8 +31,8 @@ class << model_name
     has_many :host_aggregates,               :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_networks,                :through     => :network_manager
     has_many :security_groups,               :through     => :network_manager
-
-    has_one  :source_tenant, :as => :source, :class_name => 'Tenant'
+    has_one  :source_tenant, :as => :source, :class_name  => 'Tenant'
+    has_many :vm_or_template_labels,         :through     => :vms, :source => :labels
```

Comment:
might be good to rename this to :vm_and_template_labels ? So it fits the relation.